src/stirling/source_connectors/perf_profiler/perf_profiler_bpf_test.cc

```diff
@@ -211,36 +209,36 @@ class ContainerSubProcesses final : public PerfProfilerTestSubProcesses {
 };
 
 class PerfProfileBPFTest : public ::testing::TestWithParam<std::filesystem::path> {
- public:
-  PerfProfileBPFTest()
-      : test_run_time_(FLAGS_test_run_time), data_table_(/*id*/ 0, kStackTraceTable) {}
-
  protected:
   void SetUp() override {
+    // Accepts the default profiler parameters.
+    // As compared to parameters from "SetFastTestParameters"
+    // This will have:
+    // ... a longer data transfer period,
+    // ... and a slower stack trace sample rate.
+    test_run_time_ = std::chrono::seconds{FLAGS_stirling_profiler_table_update_period_seconds};
+  }
+
+  void SetFastTestParameters() {
+    // Used by most test cases, this will cause the perf profiler to sample stack traces more
+    // frequently and transfer data from BPF maps & perf buffers, to user space, more frequently.
     FLAGS_stirling_profiler_java_symbols = true;
     FLAGS_number_attach_attempts_per_iteration = kNumSubProcs;
     FLAGS_stirling_profiler_table_update_period_seconds = 1;
     FLAGS_stirling_profiler_stack_trace_sample_period_ms = 7;
-
-    source_ = PerfProfileConnector::Create("perf_profile_connector");
-    ASSERT_OK(source_->Init());
-    source_->set_data_tables({&data_table_});
-
-    // Immediately start the transfer data thread to continue draining perf buffers,
-    // i.e. to prevent dropping perf buffer entries.
-    StartTransferDataThread();
+    test_run_time_ = std::chrono::seconds{FLAGS_test_run_time};
+    const auto period = std::chrono::seconds{FLAGS_stirling_profiler_table_update_period_seconds};
+    source_.RawPtr()->sampling_freq_mgr().set_period(period);
```

Comment:
I find it difficult to reason about these flags because we are introducing a new point in test execution where these flags can be modified.

While it seems safe to change the "fast params" after the unit connector is `Init`'ed, I think this introduces the possibility of experiencing stale reads (where a flag is read before its value is updated).

I would prefer if these test case variants were composed with Google test components (Type tests, parameterized tests. etc). That would allow these flags to be set before any application code runs and stays more consistent with our other test cases.

Reply:
Split the test cases by `FastPerfProfileBPFTest` and `SlowPerfProfileBPFTest`.